messaging/__init__.py

```diff
@@ -19,10 +19,14 @@
 
 context = Context()
 
-def new_message():
+def new_message(service=None, size=None):
   dat = log.Event.new_message()
   dat.logMonoTime = int(sec_since_boot() * 1e9)
   dat.valid = True
+  if service is not None and size is None:
```

Comment:
Can you do a nested if here? Think that would me more easy to read.

```python
if service is not None:
  if size is None:
    dat.init(service)
  else:
    data.init(service, size)
```

Reply:
Sure, let me just push the commit